--- Tk800.024/pod/options.pod	Tue Mar  5 21:47:43 2002
+++ new.Tk800.024/pod/options.pod	Mon Mar 18 14:35:43 2002
@@ -660,6 +660,15 @@
 particular widget and may be determined by other options, such as
 B<anchor> or B<justify>.
 
+=item Name:	B<tile>
+
+=item Class:	B<Tile>
+
+=item Switch:	B<-tile>
+
+Specifies image used to display the widget. If image is the empty
+string, then the normal background color is displayed.
+
 =item Name:	B<troughColor>
 
 =item Class:	B<Background>
