--- Tk800.023/Tk.xs	Wed Mar 22 13:29:59 2000
+++ new.Tk800.023/Tk.xs	Tue Mar  5 00:09:27 2002
@@ -428,6 +428,29 @@
 int	height;
 SV *	source;
 
+void
+pTk_GetBitmap(tkwin, name)
+Tk_Window	tkwin;
+char *	name;
+PPCODE:
+ {
+  Tcl_Interp *interp;
+  Pixmap pixmap;
+  if (TkToWidget(tkwin,&interp) && interp)
+   {
+    pixmap = Tk_GetBitmap(interp, tkwin, name);
+    if (pixmap == None)
+     PUSHs(&PL_sv_undef);
+    else
+     PUSHs(sv_2mortal(newSViv((IV)pixmap)));
+   }
+  else
+   {
+    croak("Invalid widget");
+   }
+ }
+
+
 MODULE = Tk	PACKAGE = Tk::Widget	PREFIX = Tk_
 
 void
