--- Tk800.024/pTk/mTk/generic/tkPlace.c	Tue Mar  5 21:47:38 2002
+++ new.Tk800.024/pTk/mTk/generic/tkPlace.c	Sun Mar 10 12:23:54 2002
@@ -272,7 +272,7 @@
 	}
 	if (slavePtr->flags & CHILD_REL_WIDTH) {
 	    Tcl_AppendElement(interp,"-relwidth");
-	    Tcl_IntResults(interp,1,1, slavePtr->relWidth);
+	    Tcl_DoubleResults(interp,1,1, slavePtr->relWidth);
 	} else {
 	    Tcl_AppendElement(interp,"-relwidth");
 	    Tcl_AppendElement(interp,"");
@@ -286,7 +286,7 @@
 	}
 	if (slavePtr->flags & CHILD_REL_HEIGHT) {
 	    Tcl_AppendElement(interp,"-relheight");
-	    Tcl_IntResults(interp,1,1, slavePtr->relHeight);
+	    Tcl_DoubleResults(interp,1,1, slavePtr->relHeight);
 	} else {
 	    Tcl_AppendElement(interp,"-relheight");
 	    Tcl_AppendElement(interp,"");
