diff -u 'Tk800.024/MANIFEST' 'new.Tk800.024/MANIFEST'
Index: ./MANIFEST
--- ./MANIFEST	Tue Mar  5 21:47:45 2002
+++ ./MANIFEST	Wed Mar 27 19:49:08 2002
@@ -332,6 +332,7 @@
 demos/demos/images/c.icon
 demos/demos/images/cursor.mask
 demos/demos/images/cursor.xbm
+demos/demos/images/cursor.cur
 demos/demos/images/dir.icon
 demos/demos/images/dir.mask
 demos/demos/images/earth.gif
@@ -1416,6 +1417,7 @@
 t/balloon.t
 t/browseentry.t
 t/create.t
+t/cursor.t
 t/dash.t
 t/fbox.t
 t/fileevent.t
diff -u 'Tk800.024/pTk/mTk/win/tkWinCursor.c' 'new.Tk800.024/pTk/mTk/win/tkWinCursor.c'
Index: ./pTk/mTk/win/tkWinCursor.c
Prereq:  1.2 
--- ./pTk/mTk/win/tkWinCursor.c	Tue Mar  5 21:47:36 2002
+++ ./pTk/mTk/win/tkWinCursor.c	Wed Mar 27 16:11:37 2002
@@ -115,6 +115,10 @@
 	cursorPtr->system = 0;
     }
     if (cursorPtr->winCursor == NULL) {
+        cursorPtr->winCursor = LoadCursorFromFile(Tk_GetHINSTANCE(), string);
+	cursorPtr->system = 0;
+    }
+    if (cursorPtr->winCursor == NULL) {
 	ckfree((char *)cursorPtr);
 	Tcl_AppendResult(interp, "bad cursor spec \"", string, "\"",
 		(char *) NULL);
diff -u /dev/null 'new.Tk800.024/t/cursor.t'
Index: ./t/cursor.t
--- ./t/cursor.t	Thu Jan  1 01:00:00 1970
+++ ./t/cursor.t	Wed Mar 27 19:51:16 2002
@@ -0,0 +1,27 @@
+# -*- perl -*-
+BEGIN { $|=1; $^W=1; }
+use strict;
+use Test;
+use Tk;
+
+BEGIN
+  {
+   plan test => 2, todo => [($Tk::platform eq 'MSWin32' ? () : (2))];
+  };
+
+my $mw;
+$mw = Tk::MainWindow->new();
+$mw->geometry('+10+10');  # This works for mwm and interactivePlacement
+
+if ($Tk::platform eq 'MSWin32') {
+    my $curfile = "demos/demos/images/cursor.cur";
+    $mw->configure(-cursor => $curfile);
+    ok(1);
+    ok($mw->cget(-cursor), $curfile);
+} else {
+    $mw->configure(-cursor => ['@demos/demos/images/cursor.xbm','black']);
+    ok(1);
+    ok(ref $mw->cget(-cursor) eq 'ARRAY');
+}
+
+__END__
