--- Tk800.024/pod/DialogBox.pod	Tue Mar  5 21:47:43 2002
+++ new.Tk800.024/pod/DialogBox.pod	Mon Mar 18 15:54:21 2002
@@ -60,7 +60,7 @@
 
 Add the widget indicated by I<widget>. I<Widget> can be the name
 of any Tk widget (standard or contributed). I<Options> are the
-options that the widget accepts. The widget is advertized as a
+options that the widget accepts. The widget is advertised as a
 subwidget of B<DialogBox>.
 
 =item B<Show(>I<grab>B<)>
@@ -71,6 +71,13 @@
 the name of the button invoked.
 
 =back
+
+=head1 ADVERTISED SUBWIDGETS
+
+The buttons are advertised by C<B_$bl>, where C<$bl> is the label of
+the button. Added widgets are advertised by the lowercase variant of
+the widget kind. If multiple widgets of the same kind are used, then
+the last widget will get the advertised name.
 
 =head1 BUGS
 
