--- Tk800.024/Tixish/DialogBox.pm	Tue Mar  5 21:47:44 2002
+++ new.Tk800.024/Tixish/DialogBox.pm	Mon Mar 18 21:13:20 2002
@@ -27,7 +27,11 @@
     $cw->{'selected_button'} = '';
     $cw->transient($cw->Parent->toplevel);
     $cw->withdraw;
-    $cw->protocol('WM_DELETE_WINDOW' => sub {});
+    if (@$buttons == 1) {
+	$cw->protocol('WM_DELETE_WINDOW' => sub { $cw->{'default_button'}->invoke });
+    } else {
+	$cw->protocol('WM_DELETE_WINDOW' => sub {});
+    }
 
     # create the two frames
     my $top = $cw->Component('Frame', 'top');
@@ -42,6 +46,7 @@
     foreach $bl (@$buttons)
      {
 	my $b = $bot->Button(-text => $bl, -command => sub { $cw->{'selected_button'} = "$bl" } );
+	$b->bind('<Return>' => [ $b, 'Invoke']);
 	$cw->Advertise("B_$bl" => $b);
         if ($Tk::platform eq 'MSWin32')
          {
@@ -56,7 +61,6 @@
 	        $b->pack(-in => $db, -padx => '2', -pady => '2');
 	        $db->pack(-side => 'left', -expand => 1, -padx => 1, -pady => 1);
             }
-	    $cw->bind('<Return>' => [ $b, 'Invoke']);
 	    $cw->{'default_button'} = $b;
 	} else {
 	    $b->pack(-side => 'left', -expand => 1,  -padx => 1, -pady => 1);
