--- Tk800.024/pod/LabFrame.pod	Tue Mar  5 21:47:43 2002
+++ new.Tk800.024/pod/LabFrame.pod	Sun Mar 24 16:07:48 2002
@@ -1,4 +1,3 @@
-
 =head1 NAME
 
 Tk::LabFrame - labeled frame.
@@ -28,6 +27,14 @@
 
 =over 4
 
+=item B<-font> =E<gt> I<font>
+
+Specifies the label's font
+
+=item B<-foreground> =E<gt> I<color>
+
+Specifies the label's foreground color
+
 =item B<-label> =E<gt> I<text>
 
 The text of the label to be placed with the Frame.
@@ -41,6 +48,33 @@
 near the northwest corner such that it appears to "overwrite" the
 groove.
 
+=item B<-labelvariable> =E<gt> I<scalar_ref>
+
+Specifies a reference to a variable and used as an 
+alternative to -label. If the value of the variable changes,
+then the widget will be updated automatically to reflect the
+change. Used like -textvariable.
+
+=back
+
+=head1 ADVERTISED SUBWIDGETS
+
+See L<Tk::mega/"Subwidget"> how to use advertised widgets.
+
+=over 4
+
+=item I<border>
+
+  Frame used to display the border
+
+=item I<frame>
+
+  Frame used to create new LabFrame children. 
+
+=item I<label>
+
+  Label used for displaying text
+
 =back
 
 =head1 EXAMPLE
@@ -74,4 +108,3 @@
 terms as Perl.
 
 =cut
-
