--- Tk800.024/Listbox/Listbox.pm	Tue Mar  5 21:48:36 2002
+++ new.Tk800.024/Listbox/Listbox.pm	Sun Mar 24 16:17:41 2002
@@ -71,6 +71,7 @@
 {
  my ($listbox,$balloon,$X,$Y,@opt) = @_;
  my $e = $listbox->XEvent;
+ return if !$e;
  my $index = $listbox->index('@' . $e->x . ',' . $e->y);
  foreach my $opt (@opt)
   {
@@ -108,7 +109,17 @@
  $mw->bind($class,'<Shift-Down>',['ExtendUpDown',1]);
 
  $mw->XscrollBind($class);
- $mw->PriorNextBind($class);
+ $mw->bind($class,'<Next>',  sub {
+	       my $w = shift;
+	       $w->yview('scroll',1,'pages');
+	       $w->activate('@0,0');
+	   });
+ $mw->bind($class,'<Prior>', sub {
+	       my $w = shift;
+	       $w->yview('scroll',-1,'pages');
+	       $w->activate('@0,0');
+	   });
+ $mw->MouseWheelBind($class);
 
  $mw->bind($class,'<Control-Home>','Cntrl_Home');
  ;
