--- Tk800.024/Tk/Menu.pm	Tue Mar  5 21:47:44 2002
+++ new.Tk800.024/Tk/Menu.pm	Sun Mar 24 16:20:47 2002
@@ -58,6 +58,7 @@
  my ($menu,$args) = @_;
  my $menuitems = delete $args->{-menuitems};
  $menu->SUPER::InitObject($args);
+ $menu->ConfigSpecs(-foreground => ['SELF']);
  if (defined $menuitems)
   {
    # If any other args do configure now
@@ -693,7 +694,7 @@
   }
  $menu->activate($i);
  $menu->GenerateMenuSelect;
- if ($menu->type($i) eq 'cascade')
+ if ($menu->cget('-type') eq 'menubar' && $menu->type($i) eq 'cascade')
   {
    my $cascade = $menu->entrycget($i, '-menu');
    $menu->postcascade($i);
