--- Tk800.024/Tk/MMutil.pm	Tue Mar  5 09:27:40 2002
+++ new.Tk800.024/Tk/MMutil.pm	Thu Aug 29 09:00:50 2002
@@ -272,6 +272,22 @@
    $self->{'LDFLAGS'} =~ s/-(debug|pdb:\w+)\s+//g;
    $self->{'LDDLFLAGS'} =~ s/-(debug|pdb:\w+)\s+//g;
   }
+ elsif ($^O eq 'darwin' )
+  {
+   $self->{'LDDLFLAGS'} =~ s/-flat_namespace//;
+   $self->{'LDDLFLAGS'} =~ s/-undefined\s+suppress//;
+   if ( -e "$Config{'archlib'}/CORE/$Config{'libperl'}" ) {
+    $self->{'LDDLFLAGS'} .= " -L\${PERL_ARCHLIB}/CORE -lperl ";
+   }
+   elsif ( -e "/System/Library/Perl/darwin/CORE/libperl.dylib" ) {
+    $self->{'LDDLFLAGS'} .= " -L/System/Library/Perl/darwin/CORE -lperl ";
+   }
+   else {
+    warn "Can't find libperl.dylib";
+   }
+   $self->{'LDFLAGS'} =~ s/-flat_namespace//;
+   $self->{'LDFLAGS'} =~ s/-undefined\s+suppress//;
+  }
  elsif ($^O =~ /(openbsd)/i)
   {
    # -Bforcearchive is bad news for Tk - we don't want all of libpTk.a in all .so-s.

