--- Tk800.024/pod/NoteBook.pod	Tue Mar  5 21:47:43 2002
+++ new.Tk800.024/pod/NoteBook.pod	Thu May  9 08:11:02 2002
@@ -142,6 +142,11 @@
 Like configure for the page indicated by I<pageName>. I<Options> may
 be any of the options accepted by the B<add> method.
 
+=item B<pages>
+
+Returns a list consisting of the names of all currently defined
+pages, i.e., those created with the B<add> method.
+
 =item B<raise(>I<pageName>B<)>
 
 Raise the page identified by I<pageName>.
