--- Tk800.024/Tk.pm	Tue Mar  5 21:48:36 2002
+++ new.Tk800.024/Tk.pm	Fri Aug  2 00:23:44 2002
@@ -136,18 +136,15 @@
 
 sub Ev
 {
- my @args = @_;
- my $obj;
- if (@args == 1)
+ if (@_ == 1)
   {
-   my $arg = pop(@args);
-   $obj = (ref $arg) ? $arg : \$arg;
+   my $arg = $_[0];
+   return bless (((ref $arg) ? $arg : \$arg), 'Tk::Ev');
   }
  else
   {
-   $obj = \@args;
+   return bless [@_],'Tk::Ev';
   }
- return bless $obj,'Tk::Ev';
 }
 
 sub InitClass
@@ -358,7 +355,7 @@
  local $_ = shift;
  unless (defined $Home)
   {
-   $Home = $ENV{'HOME'} || ($ENV{'HOMEDRIVE'}.$ENV{'HOMEPATH'});
+   $Home = $ENV{'HOME'} || (defined $ENV{'HOMEDRIVE'} && defined $ENV{'HOMEPATH'} ? $ENV{'HOMEDRIVE'}.$ENV{'HOMEPATH'} : "");
    $Home =~ s#\\#/#g;
    $Home .= '/' unless $Home =~ m#/$#;
   }
