--- Tk800.024/Tk/Wm.pm	Tue Mar  5 21:47:44 2002
+++ new.Tk800.024/Tk/Wm.pm	Sun Mar 24 18:00:33 2002
@@ -123,7 +123,10 @@
   }
  my ($X,$Y) = AnchorAdjust($w->cget('-overanchor'),$rx,$ry,$rw,$rh);
  ($X,$Y)    = AnchorAdjust($w->cget('-popanchor'),$X,$Y,-$mw,-$mh);
- $w->Post($X,$Y); 
+ # adjust to not cross screen borders
+ if ($mw > $w->screenwidth)  { $X = 0 }
+ if ($mh > $w->screenheight) { $Y = 0 }
+ $w->Post($X,$Y);
  $w->waitVisibility;
 }
 
