--- Tk800.024/t/balloon.t	Tue Mar  5 21:47:45 2002
+++ new.Tk800.024/t/balloon.t	Wed Apr 17 22:48:40 2002
@@ -4,7 +4,7 @@
 use Test;
 use Tk;
 
-BEGIN { plan tests => 8 };
+BEGIN { plan tests => 10 };
 
 my $mw = Tk::MainWindow->new;
 eval { $mw->geometry('+10+10'); };  # This works for mwm and interactivePlacement
@@ -43,6 +43,17 @@
 
 eval { $balloon->configure(-motioncommand => \&motioncmd); };
 ok($@, "", "Can't set motioncommand option");
+
+my $lb = $mw->Listbox->pack;
+$lb->insert("end",1,2,3,4);
+eval { $balloon->attach($lb, -msg => ['one','two','three','four']); };
+ok($@, "", 'Problem attaching message to Listbox items');
+
+my $slb = $mw->Scrolled('Listbox')->pack;
+$lb->insert("end",1,2,3,4);
+eval { $balloon->attach($slb->Subwidget('scrolled'),
+			-msg => ['one','two','three','four']); };
+ok($@, "", 'Problem attaching message to scrolled Listbox items');
 
 ## not yet:
 #  $l->eventGenerate("<Motion>");
